tests/project/jest.config.js

```diff
@@ -0,0 +1,22 @@
+/** @type {import('@jest/types').Config.InitialOptions} */
+const config = {
+    verbose: true,
+    testEnvironment: 'jsdom',
```

Comment:
is jsdom needed?

Reply:
It's the same as `./tests/{config,generator,utils}`. And actually, I think I was getting an error if I set it default, which is jest-circus now.

Reply:
Actually, I was just writing a note about those directories into the chatbox. It would save on tsconfig.json and jest.config.js files (and my brain) to combine those directories into ./tests/unit or something. Since they share a runtime env.

Reply:
weird, it shouldn't be calling any DOM function.

Reply:
I just tried again and the error is no more :shrug:

Reply:
Oh, this is what I'd seen:

```
FAIL  tests/utils/url.tests.ts (8.674 s)
✓ URL is enabled (69 ms)
✓ Get URL host or protocol (2 ms)
✕ Absolute URL (2 ms)
✓ Fully qualified domain (2 ms)

● Absolute URL

The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
Consider using the "jsdom" test environment.

ReferenceError: document is not defined

9 | function fixBaseURL($url: string) {
10 |     if (!anchor) {
> 11 |         anchor = document.createElement('a');
|         ^
12 |     }
13 |     anchor.href = $url;
14 |     return anchor.href;

at fixBaseURL (src/utils/url.ts:11:9)
at parseURL (src/utils/url.ts:27:31)
at getAbsoluteURL (src/utils/url.ts:41:15)
at Object.<anonymous> (tests/utils/url.tests.ts:261:26)

```

Reply:
No worries, I can fix by bringing the Dom stuff out of util and back into `css-rules.ts`

Reply:
Actually I'm going to move the test out of this environment.

Reply:
Just pushed this change: [`0d046c8` (#8266)](https://github.com/darkreader/darkreader/pull/8266/commits/0d046c8e66bb46570867bfaafd444bc2801d6253)

Reply:
Couple more errors to fix before jsdom is no longer a dependency though.

Reply:
Ok, done